### ISSUE TITLE
process.exit prevents a popup from appearing (Windows)

### ISSUE THREAD

**Issue by davidnorthetal** [ccc-tomr/simple-dialog#3]
Hey,

if I do a `process.exit()` right after I display a dialog the dialog will never appear on Windows. It seems to work fine on Linux.

```
const dia = require("dialog-node");
dia.error("ASD", "BSD", 0, () => { });
process.exit(0);
```

won't ever display an error-window.

Cheers

**Comment by ccc-tomr** [comment]
With process.exit(0) you are closing the very process that opens the dialog. It actually surprises me that the dialog stays up in Linux after your process extited.

How about calling exit in the callback function of dia.error(..)? 
Something like 
```dia.error("ASD", "BSD", 0, () => { process.exit(0) });```

This way the process only gets killed (and the dialog closed) after the user clicked on the error box.

**Comment by davidnorthetal** [comment]
sure, but it's an inconsistency between plattforms, I find the bigger issue is that I can write something on Linux that works just fine, but then on Windows it does not and the API doesn't deal with it/reflects that.

I'd still consider this a bug.

**Comment by davidnorthetal** [comment]
possibly related: https://nodejs.org/api/child_process.html#child_process_options_detached

**Comment by ccc-tomr** [comment]
I understand your logic that cross platform APIs should behave the same across the different OSs but in this case, the problem occurs outside of the API's responsibility.

I wouldn't consider it a bug/inconsistency of the API here, it's rather a problem of the implementation of the OS dialog tools on Windows versus Linux. The API does what it is expected to do, on both platforms it tries to open a dialog box. The fact that the API never gets to actually open the window on one platform because the process gets killed before the process is able to render the dialog box isn't the API's fault.

I can look into the platform specific dialog tools to see whether they can be made to behave more consistently.

**Comment by ccc-tomr** [comment]
You're right ,looks like it's how spawn is implemented slightly differently between Linux and Windows. I'll give options.detached = true a chance and see whether that makes the problem go away.

**Comment by davidnorthetal** [comment]
Hmmm... I think I've read that that would on Windows make it open its own terminal window tho?

Maybe some event can be put in nodejs event loop that nodejs will wait for before shutting down or somesuch.

**Comment by davidnorthetal** [comment]
Can you `ref` it? https://nodejs.org/api/timers.html#timers_immediate_ref

**Comment by davidnorthetal** [comment]
Or maybe spawn could be replaced with `fork` ? https://nodejs.org/api/child_process.html#child_process_child_process_fork_modulepath_args_options

**Comment by davidnorthetal** [comment]
I'll check how this looks on macos

**Comment by ccc-tomr** [comment]
Thanks, all good suggestions. I would prefer if I can make spawn behave the same across all platforms. I'll try that first and if I can't make spawn behave consistently, I will give the other suggestions a closer look.

Thanks for checking OSX

**Comment by davidnorthetal** [comment]
yeah so on High Sierra something something it behaves the same as on Linux.
The window will remain even when the process itself ends and it returns control to my terminal again.

On a related note, apparently using `process.exit` is bad practice anywho :/ https://nodejs.org/api/process.html#process_process_exit_code

**Comment by davidnorthetal** [comment]
Yeah when just using `process.exitCode` and simply foregoing using `process.exit()` it behaves fine ... guess if this is something to stay away from in nodejs anywho that could be the general solution to this issue.
